parsel/selector.py

```diff
@@ -479,9 +474,13 @@ def re(
         Passing ``replace_entities`` as ``False`` switches off these
         replacements.
         """
-        return extract_regex(
-            regex, self.get(), replace_entities=replace_entities
-        )
+        data = self.get()
+        if not isinstance(data, str):
```

Comment:
Is it backwards compatible for non-jmespath selectors, e.g. for the case selector returns nothing (and so data is None)?

Reply:
Looking at https://github.com/EchoShoot/parsel/blob/58dedc24e183dc498947fe3a91dc010278f9828a/parsel/utils.py#L58-L84, I take it the fact that `None` triggers an exception is backward-compatible, but it used to raise `TypeError` and now it is `ValueError`.

One thing we could do is remove this `if` entirely and let things fail as before. And include a test for the `None` case.

It could be argued that the new error is slightly nicer, in which case we could instead replace `ValueError` with `TypeError`. But I am fine with removing the `if` entirely, I don’t think the error messages are that much different.

Reply:
I see, thanks! I think I mixed up Selector and SelectorList.

Reply:
A different exception is not backwards compatible though. If users were handling this case via catching exceptions, their code would break. No strong opinon from me if we should make the change, but if we're making it, we should make sure it's documented as backwards incompatible.

Reply:
Although this may seem like going further down the path instead of taking the easy route... Can't we just add to the if statement to check if it's none and let it pass in such case (to break further on)?

`if data is not None and not isinstance(data,str):`

Reply:
The advantage of letting values pass altogether is that we do not break backward compatibility, things perform better (2 fewer `if` statements), and there is less code to maintain.